### DISCUSSION TITLE
What are the differences between ListPart and Named Part Bags

### DISCUSSION THREAD

**Discussion by dodyg** [OrchardCMS/OrchardCore#15906] [Q&A]
![bag-list](https://user-images.githubusercontent.com/1761249/67841431-b9f62980-fb00-11e9-834f-085bec246fd6.png)

![bag-list-2](https://user-images.githubusercontent.com/1761249/67841443-bebadd80-fb00-11e9-8985-c8f885e0bec9.png)

Common:
- They are both collections

Differences:
- You can only have one ListPart per Content Type but you can have multiple Bags

What else do I miss?

**Comment by deanmarcussen** [comment]
The big difference is 

- BagPart ContentItems are contained inside the same document as the containing ContentItem. This makes using them a lot faster, as it is one database call to retrieve everything. The disadvantage can be that you cannot have an Autoroute on them, so you can only route to the main content item, and then in that page you are able to render the BagPart content items

- ListPart loosely connects two different ContentItems together, so they can live on seperate routes.

The best example use cases of the two can be seen in TheAgencyTheme, which uses a BagPart to contain all the different content items on the main home page, whereas TheBlogTheme uses a Blog Content Item, with a ListPart of BlogPosts.

There is an open issue asking if the ListPart can be made reusable https://github.com/OrchardCMS/OrchardCore/issues/3469 and I will make a further comment on it shortly.

**Comment by sebastienros** [comment]
Related to the other question about being able to have multiple ListPart. I would suggest to create a new part called "Collection" that would be like ListPart but could be used multiple times. Because ListPart ,as it can only be selected once, can provide a nicer Admin experience, like when displaying a Blog (Admin Detail view) it shows its items. Not sure how the editor will work then to add new items. Popups are hard as it needs validation. We did it in O1 it was very hard to maintain. And only being able to select existing is what the Content Picker field is.

**Comment by sebastienros** [comment]
"It's a List" is not the same as "I have Lists" hence the distinction between ListPart and CollectionPart.

**Comment by deanmarcussen** [comment]
Good description of the difference between a ListPart and a CollectionPart Sebastien.

Here is a potential tabbed UI for the editor for a CollectionPart. See what you think
![possible collection ui](https://user-images.githubusercontent.com/13782679/68878381-85d55800-06ff-11ea-9a86-c0bf65f2fc04.PNG)

Each collection's list goes onto seperate tab.

**Comment by sebastienros** [comment]
That's an interesting way to represent them, but I would make it optional (type-part settings).

**Comment by Piedone** [comment]
See the relevant docs page: https://docs.orchardcore.net/en/latest/topics/content-relationships/
